include/pybind11/common.h

```diff
@@ -234,6 +237,54 @@ extern "C" {
     }                                                                          \
     PyObject *pybind11_init()
 
+/** \rst
+    This macro creates the entry point that will be invoked when the Python interpreter
+    imports an extension module. The module name is given as the fist argument and it
+    should not be in quotes. The second macro argument defines a variable of type
+    `py::module` which can be used to initialize the module.
+
+    .. code-block:: cpp
+
+        PYBIND11_MODULE(example, m) {
+            m.doc() = "pybind11 example module";
+
+            // Add bindings here
+            m.def("foo", []() {
+                return "Hello, World!";
+            });
+        }
+\endrst */
+#define PYBIND11_MODULE(name, variable)                                        \
+    static void pybind11_init_##name(pybind11::module &);                      \
+    PYBIND11_PLUGIN_IMPL(name) {                                               \
+        int major, minor;                                                      \
+        if (sscanf(Py_GetVersion(), "%i.%i", &major, &minor) != 2) {           \
```

Comment:
Looks like `Python.h` is guaranteed to include `<stdio.h>` (along with a few other standard headers) so we get `sscanf` anyway. Given that `std::sscanf` needs `<cstdio>`, I'd rather keep the plain version just to avoid cluttering the includes.